sw/device/tests/coverage_test.c

```diff
@@ -0,0 +1,43 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/testing/test_coverage.h"
+#include "sw/device/lib/testing/test_main.h"
+#include "sw/device/lib/uart.h"
+#include <stdbool.h>
+#include <stdint.h>
+
+static void spin_45(uint8_t state) {
```

Comment:
The sample coverage report seemed fine, but I had wondered if we should have function attributes to minimize optimizations that could mess up the coverage report. E.g. don't inline, don't unroll loops.

Reply:
Great question, I'm also not sure. I think we can decide if this is needed when we start collecting coverage from actual programs and review the reports.